enhancements/ingress/set_eip_nlb_ingress.md

```diff
@@ -0,0 +1,983 @@
+   rather than explicitly deleting the Service after changes, they can add the  `ingress.operator.openshift.io/auto-delete-load-balancer annotation`.
+   
+### 4. Any change to the service annotation does not trigger the operator to manage the load balancer service.
+  The `status.endpointPublishingStrategy.loadBalancer.providerParameters.aws.networkLoadBalancer.eipAllocations`
+  will eventually reflect the configured subnet value by mirroring the value of
+  `service.beta.kubernetes.io/aws-load-balancer-eip-allocations` on the service. There are
+  two scenarios in which the `Subnets` status won't be equal to the `Subnets` spec:
+
+  1. The cluster admin manually configured the `service.beta.kubernetes.io/aws-load-balancer-subnets`
+   annotation on the service, which is not supported and likely to cause issues.
+  2. The cluster admin configured the IngressController's `Subnets` spec, but hasn't
+   effectuated the change by deleting the service (see [section](#Cluster administrator updates eipAllocations field of the existing ingress controller)).
+
+  This proposal's use of `status.endpointPublishingStrategy` is consistent with the approach in
+  [LoadBalancer Allowed Source Ranges](/enhancements/ingress/lb-allowed-source-ranges.md),
+  but diverges with the approach in [Ingress Mutable Publishing Scope](/enhancements/ingress/mutable-publishing-scope.md).
+  The Ingress Mutable Publishing Scope design sets `status.endpointPublishingStrategy.loadBalancer.scope`
+  equal to `spec.endpointPublishingStrategy.loadBalancer.scope`, which may not always reflect the actual scope of the
+  load balancer. This proposal for `EIPs` ensures `status.endpointPublishingStrategy` reflects the _actual_ value
+  (in our case, the value of `service.beta.kubernetes.io/aws-load-balancer-eip-allocations` on the Service).
+
+  In short, the operator won't manage an annotation `service.beta.kubernetes.io/aws-load-balancer-eip-allocations` [here](https://github.com/openshift/cluster-ingress-operator/blob/master/pkg/operator/controller/ingress/load_balancer_service.go#L214-#L225)
+  Any change to the service annotation won't trigger the operator to check it's spec and then update the service annotation.
+
+### Risks and Mitigations
+
+- Providing a right number of EIPs same as the number of public subnets as per AWS guidelines is an admin
+responsibility.
+- Also providing existing available EIPs which are not assigned to any instance is important.
+- If the above two points are not followed, the Kubernetes LoadBalancer service for the IngressController
+  won't get assigned EIPs,  which will cause the LoadBalancer service to be in persistently pending state by the
+  Cloud Controller Manager (CCM). The reason for the persistently pending state is posted to the status of the IngressController.
+  As of now we are not able to identify any more risks which can affect other components of the EP.
```

Comment:
What about assigning EIPs to an internal ingress controller? Is that a risk? Will that fail a cluster? If so, should we add validation to ensure EIPAllocations are only used with External-type load balancers?

Reply:
yes. we need can assign eips only for internet facing nlb.

Reply:
I just answered here  https://github.com/openshift/enhancements/pull/1593/files#r1641646780

Reply:
I don't see any field available to set for scope in the installer-config.yaml. Also the default IC is set to [External](https://github.com/openshift/cluster-ingress-operator/blob/master/pkg)

However, we will need to check validate ingress controller CR at APP level when user provides eipAllocations and scope in the ingress controller CR.